pkg/internal/approver/manager/predicate/predicate.go

```diff
@@ -60,6 +60,22 @@ func Ready(_ context.Context, _ *cmapi.CertificateRequest, policies []policyapi.
 func SelectorIssuerRef(_ context.Context, cr *cmapi.CertificateRequest, policies []policyapi.CertificateRequestPolicy) ([]policyapi.CertificateRequestPolicy, error) {
 	var matchingPolicies []policyapi.CertificateRequestPolicy
 
+	nonEmptyOrDefault := func(s string, d string) string {
+		if s == "" {
+			return d
+		}
+		return s
+	}
+
+	// cert-manager applies controller defaults for issuer Kind and Group,
+	// which means that default values are NOT materialized in resources
+	// if omitted.
+	// So in order to make policies addressing these default values effective,
+	// we must apply cert-manager defaults on request when matching policies.
+	issKind := nonEmptyOrDefault(cr.Spec.IssuerRef.Kind, cmapi.IssuerKind)
+	issGroup := nonEmptyOrDefault(cr.Spec.IssuerRef.Group, "cert-manager.io")
+	issName := cr.Spec.IssuerRef.Name
+
```

Comment:
I was going to suggest using `scheme.Default(cr)` but that doesn't work, because cert-manager doesn't encapsulate all the resource defaults in the API...I wish it did.

```go
	s := runtime.NewScheme()
	if err := cmapi.AddToScheme(s); err != nil {
		return nil, err
	}
	s.Default(cr)
```

* https://github.com/cert-manager/cert-manager/blob/master/internal/apis/certmanager/v1/zz_generated.defaults.go

Reply:
Like this maybe, @wallrj? 😉 https://github.com/cert-manager/cert-manager/pull/7414/files